rubyClasses/decorator.rb

```diff
@@ -0,0 +1,10 @@
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
Please ensure your `decorator` class inherits the `nameable` class properly by adding `require_relative 'nameable'` to the top of your class 👍🏽